be/src/exec/vectorized/schema_scan_node.cpp

```diff
@@ -174,87 +178,77 @@ Status SchemaScanNode::get_next(RuntimeState* state, ChunkPtr* chunk, bool* eos)
     }
-    for (auto& slot_desc : dest_slot_descs) {
-        if (!slot_desc->is_materialized()) {
-            continue;
-        }
+
+    for (auto& slot_desc : src_slot_descs) {
         ColumnPtr column = vectorized::ColumnHelper::create_column(slot_desc->type(), slot_desc->is_nullable());
-        (*chunk)->append_column(std::move(column), slot_desc->id());
+        chunk_src->append_column(std::move(column), slot_desc->id());
     }
-    bool scanner_eos = false;
-    int row_num = 0;
 
-    while (true) {
-        RETURN_IF_CANCELLED(state);
+    // convert src chunk format to dest chunk format to process where clause
+    ChunkPtr chunk_dst = std::make_shared<vectorized::Chunk>();
+    if (nullptr == chunk_dst.get()) {
+        return Status::InternalError("Failed to allocate new chunk.");
+    }
 
-        if (reached_limit()) {
-            _is_finished = true;
-            // if row_num is greater than 0, in this call, eos = false, and eos will be set to true
-            // in the next call
-            if (row_num == 0) {
-                *eos = true;
-            }
-            break;
+    if (!src_slot_descs.empty()) {
+        for (size_t i = 0; i < dest_slot_descs.size(); ++i) {
+            ColumnPtr column = vectorized::ColumnHelper::create_column(dest_slot_descs[i]->type(),
+                                                                       dest_slot_descs[i]->is_nullable());
+            chunk_dst->append_column(std::move(column), dest_slot_descs[i]->id());
         }
+    }
 
-        if (row_num >= runtime_state()->chunk_size()) {
-            break;
-        }
+    while (!scanner_eos && chunk_dst->is_empty()) {
+        while (row_num < runtime_state()->chunk_size()) {
+            {
+                SCOPED_TIMER(_scanner_read_timer);
+                RETURN_IF_ERROR(_schema_scanner->get_next(&chunk_src, &scanner_eos));
+            }
 
-        ChunkPtr chunk_src = std::make_shared<vectorized::Chunk>();
-        if (nullptr == chunk_src.get()) {
-            return Status::InternalError("Failed to allocate new chunk.");
-        }
-        // init column information
-        for (auto& slot_desc : src_slot_descs) {
-            ColumnPtr column = vectorized::ColumnHelper::create_column(slot_desc->type(), slot_desc->is_nullable());
-            chunk_src->append_column(std::move(column), slot_desc->id());
-        }
-        RETURN_IF_ERROR(_schema_scanner->get_next(&chunk_src, &scanner_eos));
-
-        if (scanner_eos) {
-            _is_finished = true;
-            // if row_num is greater than 0, in this call, eos = false, and eos will be set to true
-            // in the next call
-            if (row_num == 0) {
-                *eos = true;
+            if (scanner_eos) {
+                _is_finished = true;
+                // if row_num is greater than 0, in this call, eos = false, and eos will be set to true
+                // in the next call
+                if (row_num == 0) {
```

Comment:
can condition `chunk_src->size() is less than runtime_state()->chunk_size()` is be a sign of encountering of input EOS.